docusaurus/docs/protocol/tokenomics/token_logic_modules.md

```diff
@@ -0,0 +1,301 @@
+        AA -.- TODO{See TODO below}
+    end
+
+    SM --> SD
+
+    classDef module fill:#f9f,stroke:#333,stroke-width:2px;
+    classDef address fill:#bbf,stroke:#333,stroke-width:2px;
+    classDef question fill:#e3db6d,stroke:#333,stroke-width:2px;
+    classDef event fill:#e8b761,stroke:#333,stroke-width:2px;
+
+    class SM,AM module;
+    class RSH1,RSH2,OA,OPA,AA address;
+    class TODO question;
+```
+
+:::note
+
+TODO_MAINNET: If the application stake has dropped to (near?) zero, should
+we unstake it? Should we use it's balance? Should their be a payee of last resort?
+Make sure to document whatever decision we come to.
+
+:::
+
+## TLM: Global Mint (GM)
+
+_tl;dr Distributed newly minted coins on a per claim basis to all involved stakeholders._
+
+The `Global Mint` TLM is, _theoretically_, going to reach `zero`the only when the network
+reaches maturity in the far future.
+
+On a per claim basis, the network mints new tokens based on the the amount of work
+claimed. The newly minted tokens are distributed to the supplier, DAO, service owner
+and application based on the values of various governance params.
+
+```mermaid
+---
+title: "Token Logic Module: Global Mint"
+---
+flowchart TD
+    SC(["Settlement Amount Coins (SA)"])
+    PCI(["Per Claim Global Inflation <br> (Governance Parameter)"])
+    IMC(["Inflation Mint Coins (IMC)"])
+
+    DA(["DAO Mint Allocation"])
+    PA(["Proposer Mint Allocation"])
+    SA(["Supplier Mint Allocation"])
+    SOM(["Source Owner Mint Allocation"])
+    AA(["Application Mint Allocation"])
+
+
+    SC --> IMC
+    PCI --> IMC
+    IMC --> TO
+
+    subgraph TO[Tokenomics Operations]
+        TM[[Tokenomics Module]]
+        TK[(Tokenomics Keeper)]
+        TM -..- TK
+    end
+
+
+    AA --> ID
+    DA --> ID
+    TO --Distribute Inflation (IMC)--> ID
+    PA --> ID
+    SOM --> ID
+    SA --> ID
+
+    subgraph ID[Inflation Distribution]
+        DI["Distribute Inflation (IMC)"]
+        APPA["Application Address"]
+        SPPA["Supplier Address"]
+        DAOA["DAO Address"]
+        SOA["Service Owner Address"]
+        PRA["Proposer Address"]
+
+        DI -->|% Mint Allocation| APPA
+        DI -->|% Mint Allocation| DAOA
+        DI -->|% Mint Allocation| PRA
+        DI -->|% Mint Allocation| SOA
+        DI -->|% Mint Allocation| SPPA
+    end
+
+    classDef module fill:#f9f,stroke:#333,stroke-width:2px;
+    classDef address fill:#bbf,stroke:#333,stroke-width:2px;
+    classDef question fill:#e3db6d,stroke:#333,stroke-width:2px;
+    classDef govparam fill:#eba69a,stroke:#333,stroke-width:2px;
+    classDef event fill:#e8b761,stroke:#333,stroke-width:2px;
+
+    class TM module;
+    class PCI,DA,PA,SA,SOM,AA govparam;
+    class APPA,SPPA,DAOA,SOA,PRA address;
+```
+
+## TLM: Global Mint Reimbursement Request (GMRR)
+
+_tl;dr Prevent self-dealing by over-charging applications, sending the excess to the DAO/PNF, and emitting an event as a reimbursement request._
```

Comment:
Could we charge the suppliers instead? If I'm understanding GMRR correctly this is possible.

It has the benefit of dealing with `Supplier`s which are more likely to be known and less prone to appear/disappear over time.

Reply:
We could but then we're getting into less Morse parity and more "why are you changing everything land."

We'll dive deeper into this int he coming months.